### ISSUE TITLE
Fix SC2004

### ISSUE THREAD

**Issue by noraworld** [ko1nksm/getoptions#49]
When I use getoptions, I always have to manually ignore [`SC2004`](https://www.shellcheck.net/wiki/SC2004) that [ShellCheck](https://github.com/koalaman/shellcheck) points out.

```shell
In foo line 62:
          REST="${REST} \"\${$(($OPTIND-$#))}\""
                                ^-----^ SC2004 (style): $/${} is unnecessary on arithmetic variables.


In foo line 68:
        REST="${REST} \"\${$(($OPTIND-$#))}\""
                              ^-----^ SC2004 (style): $/${} is unnecessary on arithmetic variables.

For more information:
  https://www.shellcheck.net/wiki/SC2004 -- $/${} is unnecessary on arithmeti...
```

```shell
    59	      --)
    60	        shift
    61	        while [ $# -gt 0 ]; do
*   62	          REST="${REST} \"\${$(($OPTIND-$#))}\""
    63	          shift
    64	        done
    65	        break ;;
    66	      [-]?*) set "unknown" "$1"; break ;;
    67	      *)
*   68	        REST="${REST} \"\${$(($OPTIND-$#))}\""
    69	    esac
    70	    shift
    71	  done
```

Is it possible to fix this?

**Comment by ko1nksm** [comment]
The `gengetoptions parser` has a `--shellcheck` option that generates shellcheck directives that ignore warnings. However, a bug prevented it from working properly. This bug will be fixed in the next version.

**Comment by ko1nksm** [comment]
Fixed and released v3.3.1.
